src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -487,7 +491,71 @@ private void updateBundleAndRecipientListWithResponses(
         bundle.put(question, responses);
         recipientList.put(question.getId(), recipients);
     }
-    
+
+    /**
+     * Removes instructors who are not displayed to students from the
+     * {@link FeedbackQuestionBundle}
+     * 
+     * @param bundle
+     *            {@link FeedbackQuestionsBundle} containing questions for the
+     *            session
+     */
+    private void filterOutHiddenInstructors(FeedbackSessionQuestionsBundle bundle) {
+
+        Set<String> hiddenInstructorEmails = getHiddenInstrutorEmails(bundle.getFeedbackSession().getCourseId());
```

Comment:
@unyoungwax, do you think it will be possible to rewrite this so that we only perform this database query if there are any questions with instructor recipient type?